src/main/java/com/commercetools/sync/inventory/InventorySync.java

```diff
@@ -0,0 +1,405 @@
+    protected void process(@Nonnull List<InventoryEntry> inventories) {
+        final List<InventoryEntryDraft> drafts = transformToDrafts(inventories);
+        processDrafts(drafts);
+    }
+
+    @Override
+    @Nonnull
+    public InventorySyncStatistics getStatistics() {
+        return statistics;
+    }
+
+    /**
+     * Instantiate and fill {@code supplyChannelKeyToId} with values fetched from API.
+     */
+    private void buildChannelMap() {
+        if (syncOptions.getParallelProcessing() > 1) {
+            supplyChannelKeyToId = new ConcurrentHashMap<>();
+        } else {
+            supplyChannelKeyToId = new HashMap<>();
+        }
+        inventoryService.fetchAllSupplyChannels()
+                .forEach(c -> supplyChannelKeyToId.put(c.getKey(), c.getId()));
+    }
+
+    /**
+     * Process batch in current thread in case of sequential processing or submit batch
+     * processing to {@code executorService} in case of parallel execution.
+     *
+     * @param drafts batch of {@link InventoryEntryDraft}
+     */
+    private void runDraftsProcessing(final List<InventoryEntryDraft> drafts) {
+        if (executorService != null) {
+            executorService.submit(() -> processBatch(drafts));
+        } else {
+            processBatch(drafts);
+        }
+    }
+
+    /**
+     * When processed in parallel it awaits for all tasks from {@code executorService} to end.
+     */
+    private void awaitDraftsProcessingFinished() {
+        if (executorService != null) {
+            executorService.shutdown();
+            try {
+                executorService.awaitTermination(14, TimeUnit.DAYS);
+            } catch(InterruptedException ex) {
+                LOGGER.error("Parallel inventories sync process was interrupted.", ex);
+            }
+        }
+    }
+
+    /**
+     * Method process batch of {@link InventoryEntryDraft}. The processing means:
+     * <ol>
+     *     <li>Fetching existing {@link InventoryEntry} that correspond to passed {@code drafts}</li>
+     *     <li>Deciding if each draft should be used for creating new entry or updating existing one</li>
+     *     <li>Attempting to create new entries or update existing ones</li>
+     *     <li>Updates the {@code statistics} object backed for this instance</li>
+     * </ol>
+     * @param drafts batch of drafts that need to be synced
+     */
+    private void processBatch(final List<InventoryEntryDraft> drafts) {
+        final Map<SkuKeyTuple, InventoryEntry> existingInventories = fetchExistingInventories(drafts);
+        drafts.forEach(draft -> {
+            final SkuKeyTuple skuKeyOfDraft = SkuKeyTuple.of(draft);
+            if (existingInventories.containsKey(skuKeyOfDraft)) {
+                final InventoryEntry existingEntry = existingInventories.get(skuKeyOfDraft);
+                attemptUpdate(existingEntry, draft);
+            } else {
+                attemptCreate(draft);
+            }
+            statistics.incrementProcessed();
```

Comment:
There will be single-threaded processing only from now. 100d96ff515d13f645bc086a9c10fcb46881499d